bundle-core/src/main/java/net/discdd/utils/BundleUtils.java

```diff
@@ -359,7 +363,20 @@ public static void encryptPayloadAndCreateBundle(Encrypter payloadEncryptor,
         // bundle is ready
         outerJar.close();
     }
-
+    public static Future<?> runFuture(ExecutorService executorService, String ackedEncryptedBundleId, List<ADU> adus, byte[] routingData, PipedInputStream inputPipe){
+        Future<?> future = executorService.submit(() -> {
+            PipedOutputStream outputPipe = new PipedOutputStream(inputPipe);
```

Comment:
i think we have a race condition here. try moving the new PipedOutputStream outside of the submit so that it gets created before returning.